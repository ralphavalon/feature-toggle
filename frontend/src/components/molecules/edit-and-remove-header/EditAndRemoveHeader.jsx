import React from 'react';
import { Row, Col } from 'react-bootstrap';
import { PencilSquare, XSquare } from 'react-bootstrap-icons';
import PropTypes from 'prop-types';

const EditAndRemoveHeader = ({
  headerText, onEdit, onRemove, size, className
}) => (<div className='card-header'>
  <Row>
    <Col>{headerText}</Col>
    <Col sm={size} xs={size} md={size + 1} className={className}>
      <a href="javascript:void(0)" role="button" onClick={onEdit}><PencilSquare /></a>
      <a href="javascript:void(0)" role="button" onClick={onRemove}><XSquare color="red" /></a>
    </Col>
  </Row>
</div>
);

EditAndRemoveHeader.propTypes = {
  onEdit: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired,
  headerText: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  className: PropTypes.string,
};

export default EditAndRemoveHeader;
