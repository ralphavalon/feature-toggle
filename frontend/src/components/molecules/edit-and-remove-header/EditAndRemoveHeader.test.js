import React from 'react';
import { shallow } from 'enzyme';

import EditAndRemoveHeader from './EditAndRemoveHeader';

afterEach(() => {
  jest.clearAllMocks();
});

const editAndRemoveHeaderRequiredProps = {
  onEdit: jest.fn(),
  onRemove: jest.fn(),
  headerText: 'headerText',
  size: 2
};

let editAndRemoveHeaderOptionalProps = {
  className: 'className'
};

describe('EditAndRemoveHeader', () => {
  it('renders correctly with required props', () => {
    const wrapper = shallow(<EditAndRemoveHeader {...editAndRemoveHeaderRequiredProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('renders correctly with required props and optional props', () => {
    const wrapper = shallow(<EditAndRemoveHeader {...{...editAndRemoveHeaderRequiredProps, ...editAndRemoveHeaderOptionalProps}} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should call onEdit when clicked', () => {
    const wrapper = shallow(<EditAndRemoveHeader {...editAndRemoveHeaderRequiredProps} />);
    expect(editAndRemoveHeaderRequiredProps.onEdit).toHaveBeenCalledTimes(0);
    wrapper.find('a').at(0).simulate('click');
    expect(editAndRemoveHeaderRequiredProps.onEdit).toHaveBeenCalledTimes(1);
  });
})

