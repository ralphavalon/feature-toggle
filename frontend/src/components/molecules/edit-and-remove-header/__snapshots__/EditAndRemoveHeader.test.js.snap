// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditAndRemoveHeader renders correctly with required props 1`] = `
<div
  className="card-header"
>
  <Row
    noGutters={false}
  >
    <Col>
      headerText
    </Col>
    <Col
      md={3}
      sm={2}
      xs={2}
    >
      <a
        href="javascript:void(0)"
        onClick={[MockFunction]}
        role="button"
      >
        <ForwardRef
          color="currentColor"
          size="1em"
        />
      </a>
      <a
        href="javascript:void(0)"
        onClick={[MockFunction]}
        role="button"
      >
        <ForwardRef
          color="red"
          size="1em"
        />
      </a>
    </Col>
  </Row>
</div>
`;

exports[`EditAndRemoveHeader renders correctly with required props and optional props 1`] = `
<div
  className="card-header"
>
  <Row
    noGutters={false}
  >
    <Col>
      headerText
    </Col>
    <Col
      className="className"
      md={3}
      sm={2}
      xs={2}
    >
      <a
        href="javascript:void(0)"
        onClick={[MockFunction]}
        role="button"
      >
        <ForwardRef
          color="currentColor"
          size="1em"
        />
      </a>
      <a
        href="javascript:void(0)"
        onClick={[MockFunction]}
        role="button"
      >
        <ForwardRef
          color="red"
          size="1em"
        />
      </a>
    </Col>
  </Row>
</div>
`;
